CodeReview/src/commands/CFocus.cpp

```diff
@@ -108,59 +113,45 @@ QList<Interaction::CommandSuggestion*> CFocus::suggest(Visualization::Item*, Vis
 
 CFocus::FocusInformation CFocus::extractFocusInformation(QString line)
 {
-	if (line.startsWith("[focus"))
-	{
-		auto endOfCommand = line.indexOf("]");
-		// subtract one because of starting [
-		auto commandSize = endOfCommand - 1;
-		auto commandText = line.mid(1, commandSize);
-		QStringList commandTokens = commandText.split("|");
+	QStringList commandTokens = line.split(" ");
 
-		// remove first token which is "focus"
-		commandTokens.pop_front();
+	FocusInformation focusInformation;
 
-		FocusInformation focusInformation;
+	while (!commandTokens.isEmpty())
+	{
+		auto token = commandTokens.takeFirst();
 
-		if (!commandTokens.isEmpty())
+		if (token.startsWith("h"))
+			focusInformation.type_ |= FocusInformation::FocusType::Highlight;
+		else if (token.startsWith("c"))
+			focusInformation.type_ |= FocusInformation::FocusType::Center;
+		else
 		{
-			auto token = commandTokens.takeFirst();
-			if (token.startsWith("hi"))
-				focusInformation.type_ = FocusInformation::FocusType::Highlight;
-			else if (token.startsWith("ce"))
-				focusInformation.type_ = FocusInformation::FocusType::Center;
+			bool isNumber;
+			auto step = token.toInt(&isNumber);
+
+			if (isNumber)
+			{
+				focusInformation.step_ = step;
+				break;
+			}
 		}
 
-		if (!commandTokens.isEmpty())
-		{
-			auto stepToken = commandTokens.takeFirst();
-			focusInformation.step_ = stepToken.toInt();
-		}
-		return focusInformation;
 	}
 
-	return {};
+	return focusInformation;
```

Comment:
Perhaps before returning this, you should make sure that it is valid? This is just to future proof the system. If you make the FocusInfo have `None` as type by default, and `-1` as the step. Then a valid focus info is one which has both of these different.